src/basic/path-lookup.c

```diff
@@ -721,49 +745,14 @@ char **generator_binary_paths(RuntimeScope scope) {
         return TAKE_PTR(paths);
 }
 
-char **env_generator_binary_paths(RuntimeScope runtime_scope) {
-        _cleanup_strv_free_ char **paths = NULL, **add = NULL;
-        int r;
-
-        /* First priority is whatever has been passed to us via env vars */
-        r = get_paths_from_environ("SYSTEMD_ENVIRONMENT_GENERATOR_PATH", &paths);
-        if (r < 0)
-                return NULL;
-
-        if (!paths || r > 0) {
-                switch (runtime_scope) {
-
-                case RUNTIME_SCOPE_SYSTEM:
-                        add = strv_new("/run/systemd/system-environment-generators",
-                                        "/etc/systemd/system-environment-generators",
-                                        "/usr/local/lib/systemd/system-environment-generators",
-                                        SYSTEM_ENV_GENERATOR_DIR);
-                        break;
-
-                case RUNTIME_SCOPE_USER:
-                        add = strv_new("/run/systemd/user-environment-generators",
-                                       "/etc/systemd/user-environment-generators",
-                                       "/usr/local/lib/systemd/user-environment-generators",
-                                       USER_ENV_GENERATOR_DIR);
-                        break;
-
-                default:
-                        assert_not_reached();
-                }
-                if (!add)
-                        return NULL;
-        }
-
-        if (paths) {
-                r = strv_extend_strv(&paths, add, true);
-                if (r < 0)
-                        return NULL;
-        } else
-                /* Small optimization: if paths is NULL (and it usually is), we can simply assign 'add' to it,
-                 * and don't have to copy anything */
-                paths = TAKE_PTR(add);
+char** generator_binary_paths(RuntimeScope runtime_scope) {
+        assert(IN_SET(runtime_scope, RUNTIME_SCOPE_SYSTEM, RUNTIME_SCOPE_USER));
```

Comment:
Ah, already there. I prefer to move this to _internal().

Also, why not to make this static inline?